vdr/ambassador.go

```diff
@@ -99,12 +100,21 @@ func (n *ambassador) callback(tx dag.Transaction, payload []byte) error {
 		return fmt.Errorf("callback could not process new DID Document, DID Document integrity check failed: %w", err)
 	}
 
-	isUpdate, err := n.isUpdate(nextDIDDocument)
-	if err != nil {
-		return fmt.Errorf("callback could not process new DID Document, failed to resolve current DID Document: %w", err)
+	// we check the VDR if the document is already processed by using the transaction hash
+	sourceTX := tx.Ref()
+	doc, _, err := n.didStore.Resolve(nextDIDDocument.ID, &types.ResolveMetadata{
+		SourceTransaction: &sourceTX,
+		AllowDeactivated:  true,
+	})
+	if err != nil && !errors.Is(err, types.ErrNotFound) {
+		return fmt.Errorf("could not process new DID Document: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("could not process new DID Document (tx=%s): %w", tx.Ref(), err)
```

Reply:
This callback error will already be wrapped in: `Transaction subscriber returned an error (ref=%s,type=%s)`